src/3d/p5.Renderer3D.Immediate.js

```diff
@@ -127,10 +127,9 @@ function(mode, isCurve, isBezier,isQuadratic, isContour, shapeKind){
 p5.Renderer3D.prototype._bindImmediateBuffers = function(vertices, colors){
   this._setDefaultCamera();
   var gl = this.GL;
-  var shaderKey = this._getCurShaderId();
-  var shaderProgram = this.mHash[shaderKey];
+  var shaderProgram = this._getColorVertexShader();
```

Comment:
i think we need colourVertexShader, without this change the script doesn't produce any output for the sample program:

function setup() {
  createCanvas(400, 400, WEBGL);
}
function draw() {
  background(0);
  fill(255);
  stroke(0);
  beginShape();
  vertex(0, 0, 0);
  vertex(50, 0, 0);
  vertex(50, 50, 0);
  endShape(CLOSE);
}